### PR TITLE
[Core] Fix isSet flag when setting a link to a not-yet created data

### PR DESCRIPTION
**Pull request by fredroy**
In SimpleAPI and SofaPython3 (surely XML too), 
if a data link is set to a data of a not-yet created BaseObject, `isSet` is not true.

In the newly created tests, the last one would be false without the fix in BaseData

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
Is this related to https://github.com/sofa-framework/sofa/pull/4910?

**Comment by fredroy** [comment]
> Is this related to #4910?

Maybe but It was not my intention 😅 I got the problem in SofaPython3 for our own component.


> This should fix #4911

It would have been convenient to have a test related to this problem 👼

File: Sofa/framework/SimpleApi/test/SimpleApi_test.cpp [hunk 1]
```diff
@@ -90,3 +90,45 @@ TEST_F(SimpleApi_test, createParamString )
...
+    const auto obj4 = createObject(root, "DefaultAnimationLoop", {
+        {"name", "loop4"},
+        {"printLog", "@/loop5.printLog"}
+    });
+    const auto obj5 = createObject(root, "DefaultAnimationLoop", {
+        {"name", "loop5"},
+        {"printLog", "true"}
+    });
+    auto* objdata4 = obj4->findData("printLog");
+    ASSERT_TRUE(objdata4->isSet());
```
Comment (bakpaul) [inline Sofa/framework/SimpleApi/test/SimpleApi_test.cpp:132]: Might be good to add a test with a wrong link ?

Comment (fredroy) [reply Sofa/framework/SimpleApi/test/SimpleApi_test.cpp:132]: > Might be good to add a test with a wrong link ?

done.

Actually this remark leads to interesting consequences:
 - there is no message/warning whatsoever if the link is wrong (i.e point to the non-existing data). This is understandable if setting a link _**a posteriori**_ is a feature;
 - more philosophical : should it be still isSet() if it is wrong ? but the link itself (no the pointed data) is set after all;
 - the only way to know the link was wrong is to test if `getParent()` is null... which would mean ideally that it should be tested every time...

Comment (damienmarchal) [reply Sofa/framework/SimpleApi/test/SimpleApi_test.cpp:132]: To me this is the result of a not very well defined/consistent behavior. 
Actually I'm very unconvident on the general use of links that are searched "a posteriori" and wonder if we shouldn't remove and replace that by a different (and explicit mechanism).

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Pull request merged**